docker/Dockerfile

```
@@ -1,3 +1,8 @@
 FROM alectolytic/scipy
-ADD . /code
-RUN cd /code;pip install -e .
+COPY . /code
+RUN cd /code && \
+    apk add --no-cache git && \
+    rm -rf /var/cache/apk/* && \
+    python -m pip install --upgrade --no-cache-dir pip setuptools wheel && \
+    python -m pip install --no-cache-dir -e . && \
```

Maybe we also give people the ability to use the `xmlio` extra here? That makes it a much more useful image.
```suggestion
    python -m pip install --no-cache-dir -e .[xmlio] && \
```